docs/explanation/dss-arch.rst

```diff
@@ -0,0 +1,153 @@
+DSS architecture
+================
+
+This guide provides an overview of the DSS architecture, its main components and their interactions. 
+
+Data science stack (DSS) is a ready-to-run environment for Machine Learning (ML) and Data Science (DS). 
+It's built on open-source tooling, including `MicroK8s`_, JupyterLab and `MLflow <https://ubuntu.com/blog/what-is-mlflow>`_, and usable on any Ubuntu / `snap`_-enabled workstation.
+
+Architecture overview
+---------------------
+
+DSS is a Command Line Interface (CLI)-based tool, accessible from the Ubuntu terminal. 
+See :ref:`manage_DSS` to learn about how to manage your DSS environment and the available commands.
+
+DSS is distributed as a `snap`_. 
+This provides robust security management and user-friendly version control, enabling seamless updates and auto-rollback in case of failure. 
+
+Using DSS, you can perform the following tasks: 
+
+* Installing and managing the DSS Python library.
+* Deploying and managing Jupyter Notebooks.
+* Deploying and managing MLflow.
+
+The DSS architecture involves three high-level layers:
+
+* :ref:`Application <app_layer>`.
+* :ref:`Containerisation <container_layer>`. 
+* :ref:`Operating system <os_layer>`.
+
+The following diagram showcases it:
+
+.. figure:: https://assets.ubuntu.com/v1/297b97d4-dss_arch.png
+   :width: 600px
+   :alt: text
+
+   Architecture overview
+
+The DSS square represents the application layer, MicroK8s provides the orchestration layer, and Ubuntu is the native operating system for DSS.
+More details on each layer are discussed in the following sections.
+
+.. _app_layer:
+
+Application
+~~~~~~~~~~~
+
+At the application layer, DSS includes:
+
+* Jupyter Notebooks: Open source environment that provides a flexible interface to organise DS projects and ML workloads. 
+* MLflow: Open source platform for managing ML life cycle, including experiment tracking and model registry.
+* ML frameworks: DSS comes by default with PyTorch and Tensorflow. Users can manually add other frameworks, depending on their needs and use cases.
+
+Jupyter Notebooks
+^^^^^^^^^^^^^^^^^
+
+A `Jupyter Notebook <Jupyter Notebooks_>`_ is essentially a `Kubernetes deployment <Pod_>`_, also known as `Pod`, running a Docker image with Jupyter Lab and a dedicated ML framework, such as Pytorch or Tensorflow.
+For each Jupyter Notebook, DSS mounts a `Hostpath <Microk8s hostpath docs_>`_ directory-backed persistent volume to the data directory. 
+All Jupyter Notebooks share the same persistent volume, allowing them to exchange data seamlessly. 
+
+MLflow
+^^^^^^
+
+MLflow operates in `local mode <https://mlflow.org/docs/latest/tracking.html#other-configuration-with-mlflow-tracking-server>`_, meaning that metadata and artefacts for each run are, by default, recorded to a local directory named `mlruns`. 
+This directory is backed by a persistent volume, mounted to a Hostpath directory of the MLflow Kubernetes deployment.
```

Comment:
What do we mean here by a persistent volume? Does it mean a persistent volume in the host machine, i.e., everything is stored in the local machine? This sentence is not very clear IMO. Same with the Jupyter Notebooks section.

Reply:
Local folder is translated into PV, then it is bounded to Pod through PVC.